src/docs/arch-column-row.dox

```diff
@@ -0,0 +1,93 @@
+/*! @arch_page arch-column-row Row Store & Column Store
+
+@section Definition
+
+A row oriented storage is an access method where all columns of a row are stored
+together, the data is stored by row, while a column oriented system stores its data tables by
+column. WiredTiger supports both and each management system is represented internally by a certain
+type of B-Tree data structure.
+
+Keys and values in row-stores can be anything of arbitrary length, thus, row-stores implement a
+generalized version of a B-Tree. Furthermore, keys are explicitly stored. Column-stores use a 64-bit
+unsigned integer record id as keys, thus, column-stores implement a specialized version of a B-Tree
+where the key has a predictable length. In the case of column-stores, keys are implied given the
+B-Tree position.
+
+There are two types of column-stores in WiredTiger:
+- The fixed length column-store, as its name implies, the value has a fixed length, and furthermore
+the value is restricted to 1 and 8 bits in length. The bit length is specified when the column store
+is created. The fixed length column store has specialized use cases like bitmaps.
+- The variable length is a more general case which allows for values to have any length similarly to
+row-stores.
+
+@section In-memory and on-disk format
+
+Row-stores and column-stores have a specific in-memory and on-disk data structure. Please refer to
+@ref arch-data-file to learn more.
+
+@section Row-stores
+
+Row-stores are probably seen as the traditional access method and they serve a general purpose. In
+WiredTiger, row-store keys are explicitly stored and they are usually duplicated within the same
+single file which directly impacts the on-disk storage size.
+
+@section Variable length column-stores
+
+The main advantage of variable length column-stores is on-disk storage size. Their keys are not
+stored but derived from the B-Tree location in the memory. In fact, there is a starting key on each
+page which is used to figure out the rest of the keys present on that same page. Column-stores
+values also present an advantage to further save more disk space as they can be written in more
+compact forms through encoding techniques. In WiredTiger, the run-length encoding is used to replace
+any sequence of the same value with a count and value indicator. However, this makes the access to a
+specific record more complicated, it makes impossible to jump to a record in a leaf page. Finally,
+when reading and writing column-stores, only the columns required for any particular query are
+maintained in memory.
```

Comment:
It's not quite true that `only the columns required for any particular query are maintained in memory`, that implies that you've configured the column-store to group the columns required for any particular query so that is the case. It doesn't just happen, you have to predict the queries and then create backing column-store files to reflect those queries.

Reply:
I will remove this bit, I guess I had a wrong idea about how it was working.

Reply:
See 2ab4e82d6cb995d4704fd1a97018a9f17be6d18b